### PR TITLE
fix fadvise - issue 1227

### PR DESCRIPTION
**Pull request by gjelu**
As per request in https://bugs.bareos.org/view.php?id=1227

### CODE REVIEW

File: core/src/findlib/bfile.cc [hunk 1]
```diff
@@ -1185,10 +1186,12 @@ int bclose(BareosWinFilePacket* bfd)
     bfd->cmd_plugin = false;
   } else {
 #if defined(HAVE_POSIX_FADVISE) && defined(POSIX_FADV_DONTNEED)
-    if (bfd->flags_ & O_RDONLY) {
+    /* If not RDWR or WRONLY must be Read Only */
+    if (!(bfd->m_flags & (O_RDWR|O_WRONLY))) {
       fdatasync(bfd->fid); /* sync the file */
```
Comment (franku) [inline core/src/findlib/bfile.cc:1191]:
```suggestion
```

File: core/src/findlib/bfile.cc [hunk 2]
```diff
@@ -1185,10 +1186,12 @@ int bclose(BareosWinFilePacket* bfd)
     bfd->cmd_plugin = false;
   } else {
 #if defined(HAVE_POSIX_FADVISE) && defined(POSIX_FADV_DONTNEED)
-    if (bfd->flags_ & O_RDONLY) {
+    /* If not RDWR or WRONLY must be Read Only */
+    if (!(bfd->m_flags & (O_RDWR|O_WRONLY))) {
```
Comment (arogge) [inline core/src/findlib/bfile.cc:1190]:
```suggestion
    if (!(bfd->flags_ & (O_RDWR|O_WRONLY))) {
```

File: core/src/findlib/bfile.cc (hunk 1 above)
Comment (arogge) [reply core/src/findlib/bfile.cc:1191]: We discussed this and we agree that calling `fdatasync()` is not a good idea in this case.

**Comment by arogge** [comment]
Could you please update the commit message according to https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commit-message-guideline so it at least contains `#1227: Fix fadvise bug`, so Mantis will pick up the change automatically?
Thank you!

**Pull request merged**